File: android/app/src/main/java/com/vaktinamaz/app/StepService.java
package com.vaktinamaz.app;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.Service;
import android.app.PendingIntent;

import android.content.Intent;
import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;
import android.hardware.SensorManager;

import android.os.Build;
import android.os.IBinder;
import android.util.Log;

import androidx.core.app.NotificationCompat;

import com.getcapacitor.JSObject;

// ❗ Eksik olan import EKLENDİ
import com.vaktinamaz.app.R;

public class StepService extends Service implements SensorEventListener {

    private SensorManager sensorManager;
    private Sensor stepDetector;
    private int todaySteps = 0;

    @Override
    public void onCreate() {
        super.onCreate();

        sensorManager = (SensorManager) getSystemService(SENSOR_SERVICE);

        stepDetector = sensorManager.getDefaultSensor(Sensor.TYPE_STEP_COUNTER);

        if (stepDetector == null) {
            Log.e("StepService", "Step sensor not available!");
        } else {
            sensorManager.registerListener(this, stepDetector, SensorManager.SENSOR_DELAY_NORMAL);
        }

        createNotification();
    }

    private void createNotification() {
        String channelId = "step_counter_channel";

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel channel = new NotificationChannel(
                channelId,
                "Step Counter",
                NotificationManager.IMPORTANCE_LOW
            );

            NotificationManager manager = getSystemService(NotificationManager.class);
            manager.createNotificationChannel(channel);
        }

        NotificationCompat.Builder notification = new NotificationCompat.Builder(this, channelId)
            .setContentTitle("Adım Sayar Çalışıyor")
            .setContentText("Arka planda adımlar takip ediliyor")
            .setSmallIcon(R.mipmap.ic_launcher);

        startForeground(1, notification.build());
    }

    @Override
    public void onSensorChanged(SensorEvent event) {
        todaySteps = (int) event.values[0];

        Log.d("StepService", "Adım güncellendi: " + todaySteps);

        JSObject data = new JSObject();
        data.put("stepCount", todaySteps);

        StepCounterPlugin.sendStepEvent(data);
    }

    @Override
    public void onAccuracyChanged(Sensor sensor, int accuracy) {}

    @Override
    public IBinder onBind(Intent intent) { return null; }

    @Override
    public void onDestroy() {
        super.onDestroy();
        sensorManager.unregisterListener(this);
    }
}

File: android/app/src/main/java/com/vaktinamaz/app/StepService.kt
package com.vaktinamaz.app

import android.app.*
import android.content.Context
import android.content.Intent
import android.hardware.Sensor
import android.hardware.SensorEvent
import android.hardware.SensorEventListener
import android.hardware.SensorManager
import androidx.core.app.NotificationCompat
import com.getcapacitor.JSObject
import com.getcapacitor.Bridge

class StepService : Service(), SensorEventListener {

    private lateinit var sensorManager: SensorManager
    private var stepSensor: Sensor? = null
    private var lastValue = 0

    companion object {
        const val CHANNEL_ID = "step_counter_channel"
        var bridge: Bridge? = null
    }

    override fun onCreate() {
        super.onCreate()

        sensorManager = getSystemService(Context.SENSOR_SERVICE) as SensorManager
        stepSensor = sensorManager.getDefaultSensor(Sensor.TYPE_STEP_COUNTER)

        createNotificationChannel()

        startForeground(
            1001,
            NotificationCompat.Builder(this, CHANNEL_ID)
                .setContentTitle("Adım Sayar Çalışıyor")
                .setContentText("Arka planda adımlar takip ediliyor")
                .setSmallIcon(android.R.drawable.ic_media_play)
                .setOngoing(true)
                .build()
        )

        if (stepSensor != null) {
            sensorManager.registerListener(this, stepSensor, SensorManager.SENSOR_DELAY_NORMAL)
        }
    }

    override fun onDestroy() {
        sensorManager.unregisterListener(this)
        super.onDestroy()
    }

    override fun onBind(intent: Intent?) = null

    override fun onSensorChanged(event: SensorEvent?) {
        event?.let {
            val current = it.values[0].toInt()

            if (lastValue == 0) {
                lastValue = current
                return
            }

            val diff = current - lastValue
            lastValue = current

            if (diff > 0) {
                val data = JSObject().put("steps", diff)
                bridge?.plugin("StepServicePlugin")
                    ?.notifyListeners("stepUpdate", data)
            }
        }
    }

    override fun onAccuracyChanged(sensor: Sensor?, accuracy: Int) {}

    private fun createNotificationChannel() {
        val channel = NotificationChannel(
            CHANNEL_ID,
            "Adım Sayar Servisi",
            NotificationManager.IMPORTANCE_LOW
        )
        val manager = getSystemService(NotificationManager::class.java)
        manager.createNotificationChannel(channel)
    }
